dbos_transact/dbos.py

```diff
@@ -610,10 +749,67 @@ def temp_wf(*args: Any, **kwargs: Any) -> Any:
             set_dbos_func_name(temp_wf, "<temp>." + func.__qualname__)
             self.register_wf_function(get_dbos_func_name(temp_wf), wrapped_wf)
 
-            return cast(Communicator, wrapper)
+            return cast(Communicator[R], wrapper)
 
         return decorator
 
+    @staticmethod
+    def default_required_roles(roles: List[str]) -> Callable[[Type[Any]], Type[Any]]:
+        def set_roles(cls: Type[Any]) -> Type[Any]:
+            ci = get_or_create_class_info(cls)
+            ci.def_required_roles = roles
+            return cls
+
+        return set_roles
+
+    @staticmethod
+    def check_required_roles(
```

Comment:
I know we're not consistent about this, but we should give this an underscore prefix to mark it as private